### PR TITLE
feat(argo-cd): Ability to provide cluster role to repo-server

### PR DESCRIPTION
**Pull request by ahalay**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by mkilchhofer** [comment]
Hi @ahalay 👋
Can you explain the use case for this feature?

**Review by mkilchhofer — COMMENTED** [review]
Hi @ahalay 👋
Can you explain the use case for this feature? I don't think reposerver needs cluster access for whatever use case ;-)

**Comment by ahalay** [comment]
Hello @mkilchhofer 
In my case, it is for the custom helmfile plugin. Some helmfile hooks contain a logic with access to Kubernetes resources. But it probably would be useful for any custom plugin with similar behavior.

**Comment by mkilchhofer** [comment]
> Hello @mkilchhofer
> In my case, it is for the custom helmfile plugin. Some helmfile hooks contain a logic with access to Kubernetes resources. But it probably would be useful for any custom plugin with similar behavior.

Okay 👌  this seems like a valid use case :-) Thanks for explaining that. I will have a look into your changes.

File: charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{{- if .Values.repoServer.serviceAccount.create }}
+{{- if .Values.repoServer.clusterAdminAccess.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "argo-cd.repoServer.fullname" . }}
+  labels:
+    {{- include "argo-cd.labels" (dict "context" . "component" .Values.repoServer.name "name" .Values.repoServer.name) | nindent 4 }}
+rules:
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml:9]: What do you think about reusing what we use for the controller?
https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/values.yaml#L222-L229

File: charts/argo-cd/templates/argocd-repo-server/clusterrolebinding.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.repoServer.serviceAccount.create }}
+{{- if .Values.repoServer.clusterAdminAccess.enabled }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-repo-server/clusterrolebinding.yaml:2]: Use `and`

File: charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{{- if .Values.repoServer.serviceAccount.create }}
+{{- if .Values.repoServer.clusterAdminAccess.enabled }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml:2]: Use `and`

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

File: charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml (hunk 1 above)
Comment (ahalay) [reply charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml:8]: I hope I understood your idea correctly 😅

**Review by ahalay — COMMENTED** [review]
Requested changes have been made

File: charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml (hunk 1 above)
Comment (mkilchhofer) [reply charts/argo-cd/templates/argocd-repo-server/clusterrole.yaml:8]: yep, looks good. will have a deeper look into it as soon as possible

**Review by mkilchhofer — APPROVED** [review]
lgtm. Thank you for implementing my requested changes 🙏😎

**Comment by ErikLundJensen** [comment]
I have just seen the implementation. I would really have liked two flags for creating ClusterRole and ClusterRoleBinding.
Often you would like the ClusterRole, however, create RoleBinding for each namespace that ArgoCD operates on.
In new implementation we can't op-out of the ClusterRoleBinding.

**Pull request merged**